src/librustc/ty/layout.rs

```diff
@@ -1087,25 +1089,33 @@ impl<'a, 'gcx, 'tcx> Layout {
             // Arrays and slices.
             ty::TyArray(element, count) => {
                 let element = element.layout(infcx)?;
+                let element_size = element.size(dl);
+                let count = count as u64;
```

Comment:
if we care, maybe make `fn usize_to_u64(x: usize)` and then do `usize_to_u64(count)`?